prefect_kubernetes/utilities.py

```diff
@@ -0,0 +1,73 @@
+""" Utilities for working with the Python Kubernetes API. """
+from pathlib import Path
+from typing import TypeVar, Union
+
+from kubernetes.client import models as k8s_models
+from prefect.infrastructure.kubernetes import KubernetesJob, KubernetesManifest
+
+base_types = {"str", "int", "float", "bool", "list[str]", "dict(str, str)", "object"}
+
+V1KubernetesModel = TypeVar("V1KubernetesModel")
+
+
+def convert_manifest_to_model(
+    manifest: Union[Path, KubernetesManifest], v1_model_name: str
+) -> V1KubernetesModel:
+    """Recursively converts a `dict` representation of a Kubernetes resource to the
+    corresponding Python model containing the Python models that compose it,
+    according to the `openapi_types` on the class retrieved with `v1_model_name`.
+
+    If `manifest` is a path-like object with a `.yaml` or `.yml` extension, it will be
+    treated as a path to a Kubernetes resource manifest and loaded into a `dict`.
+
+    Args:
+        manifest: A path to a Kubernetes resource manifest or its `dict` representation.
+        v1_model_name: The name of a Kubernetes client model to convert the manifest to.
+
+    Returns:
+        A populated instance of a Kubernetes client model with type `v1_model_name`.
+
+    Raises:
+        ValueError: If `v1_model_name` is not a valid Kubernetes client model name.
+        ValueError: If `manifest` is path-like and is not a valid yaml filename.
+    """
+    if not manifest:
+        return None
+
+    if not (isinstance(v1_model_name, str) and v1_model_name in set(dir(k8s_models))):
```

Comment:
Do we need to check if it's string if the type hinting is str? If so, can we break down this line into two variables:
`is_string` and `is_model_name` for readability